src/cmd/ksh93/bltins/typeset.c

```diff
@@ -1102,7 +1102,6 @@ int	b_builtin(int argc,char *argv[],Shbltin_t *context)
 		sh_addlib(tdata.sh,library,arg,NiL);
 	}
 	else
-#endif /* SHOPT_DYNAMIC */
```

Comment:
It's going to take me a couple of days to review this PR and test it on various platforms. But I'm confused by this particular set of changes. Why would switching from Nmake to Meson necessitate changing which blocks of code are predicated on the build time `SHOPT_DYNAMIC` symbol?

Also, it looks like this flag defaults to one (under Nmake) and there is no feature detection that sets it to zero. So it is presumably expected to be set via an environment variable to zero if the person building ksh does not want to support dynamic loading of builtins. It's not obvious we should continue to support that as a build time option. It would probably be better to just automatically enable/disable the feature based on the capabilities of the platform. And today it's likely to always be enabled for any platform we're targeting.

Reply:
By default ksh compiles with ```SHOPT_DYNAMIC``` option. I believe compilation was not tested when this flag is off so the ```ifdefs``` were misplaced.

Reply:
@siteshwar, Are you saying this step in switching to Meson does not set the `SHOPT_DYNAMIC` symbol? If so this portion of the change may be perfectly reasonable. But I can't help but wonder if we shouldn't instead just set the symbol to one to mimic what happens by default under Nmake (i.e., *bin/package build*). Then defer fixing the conditional compilation logic to a separate update. Alternatively we should create a file or Github issue where we track these types of deviations from the established defaults and just do whatever makes the transition to Meson the easiest in the short term.

Reply:
Compiling with this option has a dependency on ```libdll```. So for now I kept it ```off```. I would prefer to investigate later if we can remove dependency over ```libdll``` (or choose to compile it).

Reply:
and "by default" I meant the legacy build system compiles it.

Reply:
> Compiling with this option has a dependency on libdll. So for now I kept it off. 

Seems reasonable. It's better to do this transition to Meson incrementally rather than try to make everything work with the first commit. But a "TODO" comment to remind us to revisit this might be a good idea.

Reply:
I have added TODO notes.